source/DSCResources/DSC_CMBoundaryGroups/DSC_CMBoundaryGroups.psm1

```diff
@@ -0,0 +1,364 @@
+
+    .PARAMETER SiteCode
+        Specifies the SiteCode for the Configuration Manager site.
+
+    .PARAMETER BoundaryGroup
+        Specifies the boundary group name.
+
+    .Parameter Boundaries
+        Specifies an array of boundaries that must be identical in the boundary group.
+
+    .Parameter BoundariesToInclude
+        Specifies an array of boundaries to append to the boundary group.
+        If boundaries are specified this setting is ignored.
+
+    .Parameter BoundariesToExclude
+        Specifies an array of boundaries must be absent from the boundary group.
+        If boundaries are specified this setting is ignored.
+
+    .Parameter Ensure
+        Specifies if the boundary group is to be absent or present.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $BoundaryGroup,
+
+        [Parameter()]
+        [String[]]
+        $Boundaries,
+
+        [Parameter()]
+        [String[]]
+        $BoundariesToInclude,
+
+        [Parameter()]
+        [String[]]
+        $BoundariesToExclude,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -BoundaryGroup $BoundaryGroup
+
+        if ($Ensure -eq 'Present')
+        {
+            if ($state.Ensure -eq 'Absent')
+            {
+                Write-Verbose -Message ($script:localizedData.CreateBoundaryGroup -f $BoundaryGroup)
+                New-CMBoundaryGroup -Name $BoundaryGroup
+            }
+
+            if (($Boundaries) -or ($BoundariesToInclude))
+            {
+                if ($Boundaries)
+                {
+                    $includes = $Boundaries
+                }
+                else
+                {
+                    $includes = $BoundariesToInclude
+                }
+
+                foreach ($member in $includes)
+                {
+                    if (Get-CMBoundary -BoundaryName $member)
+                    {
+                        if ($state.Boundaries -notcontains $member)
+                        {
+                            Write-Verbose -Message ($script:localizedData.AddingBoundary -f $BoundaryGroup, $member)
+                            Add-CMBoundaryToGroup -BoundaryName $member -BoundaryGroupName $BoundaryGroup
+                        }
+                    }
+                    else
+                    {
+                        $errorMsg += ($script:localizedData.BoundaryAbsent -f $member)
+                        Write-Verbose -Message ($script:localizedData.BoundaryAbsent -f $member)
+                    }
+                }
+            }
+
+            if (($Boundaries) -or ($BoundariesToExclude))
```

Comment:
This: (($Boundaries) -or ($BoundariesToExclude))
Can be: ($Boundaries -or $BoundariesToExclude)